### PR TITLE
feature(site): Add Case Studies (List View, Case Study Pages)

### LINKED ISSUE #382: Case Study MVP Implementation
Implement pages related to Case Studies MVP

Implementation should be based on the [Figma Design](https://www.figma.com/file/bJsG3hEp3zsDdKgZJARPWxL7/Portfolio-Items?node-id=1%3A2)

Acceptance Criteria:

- [x] #383  
- [x] #384 
- [ ] #385
- [ ] #386 
- [ ] #387 
- [x] #389 
- [x] #390

### PR DESCRIPTION
**Pull request by FrankyMartz**
## Overview

Add Case Study pages and Case Study List page to site. The `case-studies-feature-branch` had become stale, so `master` merge required manual patching and hand-rolled resolution.

_In order to retain commit updates made to  (`BlogHeader`, `PostExcerpt`), without losing case study changes made to renamed versions of those components (`PageHeader`,  `ContentSummary`); I've kept both versions and isolated  (`PageHeader`,  `ContentSummary`) to case studies._

**Please Review 👇** 

### Patch Explanation

The `case-studies-feature-branch` branch renamed (`BlogHeader`, `PostExcerpt`) to (`PageHeader`,  `ContentSummary`) respectively. However, Git interpreted these actions as (delete / add), and no patches from merge could be consolidated from `master` to those components. — i.e. patches to those components would be lost without manual consolidation.

### Merge Patch

- Add back components `BlogHeader`, `PostExcerpt`
- Isolate components `PageHeader`,  `ContentSummary` to Case Studies
- Revert blog post pages to `BlogHeader`, `PostExcerpt`

### Original Updates

- Add Case Study Web Pages
  - List View for Case Studies — `/case-studies`
  - Detail View for Case Study — `/case-studies/<case-study-slug>`
- Add Generic component
  - `/components/molecules/PageHeader.jsx`
  - `/components/molecules/ContentSummary.jsx`

## Test Plan

Smoke Test conducted on local machine. Ideal secondary review recommended.

### Validation
- [x] Blog Post List Page
- [x] Blog Post Detail Page
- [x] Case Study List Page
- [x] Case Study Detail Page

Screen Shot 2020-02-24 at 9 23 05 AM

Screen Shot 2020-02-24 at 9 24 06 AM

Screen Shot 2020-02-24 at 9 24 21 AM

## Closing Issues
closes #382

### CODE REVIEW

**Comment by zachcd** [comment]
Edge is having a sizing issue with the images in the Case Studies List page
image

Compare to Firefox:
image

Chrome also does something strange but that's related to Gatsby Image and I don't believe it will continue into production.
image

File: src/templates/post.jsx [hunk 1]
```diff
@@ -27,6 +27,25 @@ const PostContent = styled(Section)`
   `};
 `;
 
+// NOTE: elements left over from case-studies-feature-branch merge
```
[the remark this answers is not in the document]
Comment (FrankyMartz) [reply src/templates/post.jsx]: Updates are Pushed:
* Removed redundant commented cruft
* Patched image distortion
* Patcher Header Title/Image squashing in IE/Edge

**Comment by FrankyMartz** [comment]
@zachcd I wasn't able to recreate the image degradation that was happening in your Chrome. However, I removed flexbox image wrappers for those components and wonder if that would have help.

### Validation in OS:

**MacOS:** _Chrome, FireFox, Safari_
**Microsoft:** _IE11, EDGE,_

**Comment by zachcd** [comment]
@FrankyMartz That Chrome issue has happened to me on other branches, but doesn't happen to others.  I don't think its an issue you should worry about.  It won't be like that in production.

**Pull request merged**